brave/brave-core
# Adds table view for excluded sites in Rewards Settings

Fixes: https://github.com/brave/brave-browser/issues/3657
UI PR: https://github.com/brave/brave-ui/pull/443

A lot of code was removed related to getting the excluded publishers length number, now we just check the length of the excluded ContentSiteList we get back from the publisher_db.

<img width="628" alt="Screen Shot 2019-04-10 at 4 26 54 PM" src="https://user-images.githubusercontent.com/8732757/55920602-5b8e4b80-5bae-11e9-951e-2d8201f378c7.png">

<img width="721" alt="Screen Shot 2019-04-10 at 4 27 07 PM" src="https://user-images.githubusercontent.com/8732757/55920605-5d580f00-5bae-11e9-9f0b-ec757046018b.png">

<img width="721" alt="Screen Shot 2019-04-10 at 4 27 17 PM" src="https://user-images.githubusercontent.com/8732757/55920606-5e893c00-5bae-11e9-8eca-5a7a7c27f162.png">


## Submitter Checklist:

- Verified that these changes build without errors on
- Verified that these changes pass automated tests (`npm test brave_unit_tests && npm test brave_browser_tests && npm run test-security`) on

## Test Plan:
1. Start Brave, enable Rewards.
2. Add a few sites to the AC table.
3. Exclude a single site
4. Confirm that the "Show All" button is present in the lower right hand corner of the settings box.
5. Click on it and confirm that the three AC sites are in the supported sites table and that the single excluded site is in the excluded table.
6. Under Supported Sites in the modal, remove another site.
7. Confirm the excluded count increments and the site is now present in that table.
8. Confirm that under the excluded view that restoring a site/restore all removes the respective sites from the excluded table.
9. Confirm that any restored sites can be re-added to the AC table.

## Reviewer Checklist:
